os-eap-launch/added/launch/security-domains.sh

```diff
@@ -5,17 +5,77 @@ function prepareEnv() {
   unset SECDOMAIN_ROLES_PROPERTIES
   unset SECDOMAIN_LOGIN_MODULE
   unset SECDOMAIN_PASSWORD_STACKING
+
+  for prefix in $(echo $SECURITY_DOMAINS | sed "s/,/ /g"); do
+    clearDomainEnv $prefix
+  done
+  unset SECURITY_DOMAINS
+
+}
+
+function clearDomainEnv() {
+  local prefix=$1
+
+  unset ${prefix}_LOGIN_MODULE_CODE
+  unset ${prefix}_LOGIN_MODULE_MODULE
+
+  for option in $(compgen -v | grep -s "${prefix}_MODULE_OPTION_"); do
+    unset ${option}
+  done
 }
 
 function configure() {
+  configure_legacy_security_domains
```

Comment:
Another option is to define mappings in backward-compatibility.sh using variable expression trickery.  Just an idea.